src/test/java/uk/gov/pay/api/resources/directdebit/DirectDebitResourceITBase.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.pay.api.resources.directdebit;
+
+import com.github.tomakehurst.wiremock.junit.WireMockClassRule;
+import com.spotify.docker.client.exceptions.DockerException;
+import io.dropwizard.testing.junit.DropwizardAppRule;
+import io.restassured.response.ValidatableResponse;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import uk.gov.pay.api.app.PublicApi;
+import uk.gov.pay.api.app.config.PublicApiConfig;
+import uk.gov.pay.api.it.rule.RedisDockerRule;
+import uk.gov.pay.api.utils.ApiKeyGenerator;
+import uk.gov.pay.api.utils.PublicAuthMockClient;
+import uk.gov.pay.api.utils.mocks.ConnectorDDMockClient;
+
+import static io.dropwizard.testing.ConfigOverride.config;
+import static io.dropwizard.testing.ResourceHelpers.resourceFilePath;
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static javax.ws.rs.core.HttpHeaders.AUTHORIZATION;
+import static uk.gov.pay.api.model.TokenPaymentType.DIRECT_DEBIT;
+import static uk.gov.pay.commons.testing.port.PortFactory.findFreePort;
+
+public class DirectDebitResourceITBase {
```

Comment:
Extracted this out from DirectDebitPaymentsResourceTest. Can be used by other DD Resource tests